test/cloudtest/test_k8s_node_recovery.py

```diff
@@ -0,0 +1,238 @@
+    reached_index = get_current_counter_index(mz, cluster_setup)
+
+    suspend_first_node_of_pod(mz, cluster_setup.storage_name)
+
+    # expect (stale) data to be delivered timely
+    validate_state(
+        mz,
+        cluster_setup,
+        reached_index,
+        must_exceed_reached_index=False,
+        timeout_in_sec=30,
+    )
+
+    print("Expecting stalled data...")
+    # only request this index because the previous validation succeeded / did not block
+    stalled_index = get_current_counter_index(mz, cluster_setup)
+
+    print("Awaiting node recovery...")
+    # expect live data to be delivered at most after two minutes in production (or 300s in k8s)
+    validate_state(
+        mz,
+        cluster_setup,
+        stalled_index,
+        must_exceed_reached_index=True,
+        timeout_in_sec=330,
+    )
+
+
+@pytest.mark.long
+def test_unreplicated_compute_cluster_on_failing_node(
+    mz: MaterializeApplication,
+) -> None:
+    """
+    An unreplicated compute cluster is on the failed node. Queries of indexes on the compute cluster should fail, but
+    resolve within a minute or two.
+    """
+    cluster_setup = ClusterSetup(2)
+    populate(mz, cluster_setup)
+    reached_index = get_current_counter_index(mz, cluster_setup)
+
+    suspend_first_node_of_pod(mz, cluster_setup.compute_name)
+
+    print("Awaiting node recovery...")
+    # expect (live) data to be delivered after at most after two minutes in production (or 300s in k8s)
+    validate_state(
+        mz,
+        cluster_setup,
+        reached_index + 1,
+        must_exceed_reached_index=True,
+        timeout_in_sec=330,
+    )
+
+
+@pytest.mark.long
+def test_replicated_compute_cluster_on_failing_node(mz: MaterializeApplication) -> None:
+    """
+    A replicated compute cluster is on the failed node. Queries of indexes on the compute cluster should experience no
+    disruption in latency, thanks to the second replica.
+    """
+    cluster_setup = ClusterSetup(3, compute_replication_factor=2)
+    populate(mz, cluster_setup)
+    reached_index = get_current_counter_index(mz, cluster_setup)
+
+    suspend_first_node_of_pod(mz, cluster_setup.compute_name)
+
+    print("Awaiting node recovery...")
+    # expect (live) data to be delivered without disruption in latency
+    validate_state(
+        mz,
+        cluster_setup,
+        reached_index + 1,
+        must_exceed_reached_index=True,
+        timeout_in_sec=30,
+    )
+
+
+@pytest.mark.long
+def test_envd_on_failing_node(mz: MaterializeApplication) -> None:
+    """
+    environmentd is on the failed node. All connections should fail, but resolve within a minute or two.
+    """
+    cluster_setup = ClusterSetup(4)
+    populate(mz, cluster_setup)
+    reached_index = get_current_counter_index(mz, cluster_setup)
+
+    envd_node_name = mz.get_k8s_value(
+        "app=environmentd", "{.items[*].spec.nodeName}", remove_quotes=True
+    )
+    mz.suspend_k8s_node(envd_node_name)
```

Comment:
I think for all suspends you also need to have a resume, for the following two reasons:
1. So that you can check yet again after the resume if the source has made yet more progress. This will verify that an old clusterd node that suddenly came back up does not perturb neither environmentd nor the newly-spawned clusterd.
2. So that the KinD cluster gets back to its original state of having 2 nodes in the first availability zone, allowing for the next test in the sequence to run and succeed.

Reply:
✅ Addressed in https://github.com/MaterializeInc/materialize/pull/20491/commits/e4cdf32abe245488f400b46669de5cf2b36cf19e.